cmdstanpy/stanfit.py

```diff
@@ -509,61 +508,47 @@ def num_unconstrained_params(self) -> int:
         however a model with variables ``real alpha`` and ``simplex[3] beta``
         has 4 constrained and 3 unconstrained parameters.
         """
-        if not self._validate_csv and self._metadata is None:
-            self._logger.warning(
-                'csv files not yet validated, run method validate_csv_files()'
-                ' in order to retrieve sample metadata.'
-            )
-            return None
+        if self._metadata is None:
+            self.validate_csv_files()
         return self._metadata.cmdstan_config['num_unconstrained_params']
 
     @property
     def sampler_config(self) -> Dict:
         """Returns dict of CmdStan configuration arguments."""
+        if self._metadata is None:
+            self.validate_csv_files()
         return self._metadata.cmdstan_config
 
     @property
     def sampler_vars_cols(self) -> Dict:
         """
         Returns map from sampler variable names to column indices.
         """
-        if not self._validate_csv and self._metadata is None:
-            self._logger.warning(
-                'csv files not yet validated, run method validate_csv_files()'
-                ' in order to retrieve sample metadata.'
-            )
-            return None
+        if self._metadata is None:
+            self.validate_csv_files()
         return self._metadata.sampler_vars_cols
 
+    @property
+    def stan_vars_cols(self) -> Dict:
```

Comment:
I realize that there is a large body of code in the package which _doesn't_ do this, but the type hint convention is to specify the types in the dictionary (or other collection). For example, I believe the full return type of this function would be `Dict[str, Tuple[int]]`. This may be worth a separate PR to fix up all the existing types, which I'd be happy to do if we wanted